elfi/model/elfi_model.py

```diff
@@ -458,33 +461,52 @@ def _give_name(self, name, model):
                 name = self._new_name(name[:-1], model)
             return name
 
+        try:
+            name = self._inspect_name()
+        except:
```

Comment:
Yes, but in this case it's hard to know what it would throw. Also this is in no way critical if it fails, we just let it fail.